viamonvif/discovery.go

```diff
@@ -1,241 +1,120 @@
 // Package viamonvif provides ONVIF integration to the viamrtsp module
+//nolint: revive
 package viamonvif
 
 import (
-	"bytes"
 	"context"
 	"encoding/xml"
-	"errors"
 	"fmt"
-	"io"
 	"net"
-	"net/http"
 	"net/url"
-	"slices"
 	"strings"
-	"sync"
 
-	"github.com/use-go/onvif"
-	"github.com/use-go/onvif/device"
-	"github.com/use-go/onvif/media"
-	wsdiscovery "github.com/use-go/onvif/ws-discovery"
-	onvifxsd "github.com/use-go/onvif/xsd/onvif"
+	"github.com/viam-modules/viamrtsp/viamonvif/device"
+	"github.com/viam-modules/viamrtsp/viamonvif/xsd/onvif"
 	"go.viam.com/rdk/logging"
-	"go.viam.com/utils"
-)
-
-const (
-	streamTypeRTPUnicast = "RTP-Unicast"
 )
 
 // OnvifDevice is an interface to abstract device methods used in the code.
 // Used instead of onvif.Device to allow for mocking in tests.
 type OnvifDevice interface {
-	CallMethod(request interface{}) (*http.Response, error)
+	GetDeviceInformation() (device.GetDeviceInformationResponse, error)
+	GetProfiles() (device.GetProfilesResponse, error)
+	GetStreamURI(profile onvif.Profile, creds device.Credentials) (*url.URL, error)
 }
 
-// callAndParse sends a request to an ONVIF device, decoding and mutating the
-// provided response struct with the result. Returns an error on failure.
-func callAndParse(logger logging.Logger, deviceInstance OnvifDevice, request interface{}, response any) error {
-	resp, err := deviceInstance.CallMethod(request)
-	if err != nil {
-		return err
-	}
-	defer resp.Body.Close()
+// DiscoverCameras performs WS-Discovery
+// then uses ONVIF queries to get available RTSP addresses and supplementary info.
+func DiscoverCameras(creds []device.Credentials, manualXAddrs []*url.URL, logger logging.Logger) (*CameraInfoList, error) {
```

Comment:
I changed the signature to take a list of credentials and to allow the caller to specify a list of hosts (ip:port) to attempt to discover rtsp streams from using onvif. This allows the option in the future for a caller to attempt to detect onvif rtsp streams even if WS-Discovery is not working for that devcie.